Usbasploader bootloader option addition (#6304)
qmk/qmk_firmware: docs/flashing.md

3. Flash a .hex file
4. Reset the device into application mode (may be done automatically)

## USBasploader

> USBasploader is a bootloader developed by matrixstorm. It is used in some atmega328p or similar devices which only function using the VUSB.

Reviewer: ```suggestion
USBasploader is a bootloader developed by matrixstorm. It is used in some non-USB AVR chips such as the ATmega328P, which run V-USB.
```

> To ensure compatibility with the USBasploader bootloader, make sure this block is present your `rules.mk`:

Reviewer: ```suggestion
To ensure compatibility with the USBasploader bootloader, make sure this block is present in your `rules.mk`:
```

>     # Bootloader

Reviewer: This block should be triple-backticked, so it can be syntax highlighted.

Author: I followed what the other sections followed. which is not triple-backtick

Reviewer 2: @fauxpark I think this can be put off to later. I want to do a big overhaul of the flashing docs (not just this file), once both this, and my `:flash` target PR get merged.

    #     This definition is optional, and if your keyboard supports multiple bootloaders of
    #     different sizes, comment this out, and the correct address will be loaded
    #     automatically (+60). See bootloader.mk for all options.
    BOOTLOADER = USBasp

Compatible flashers:

* [QMK Toolbox](https://github.com/qmk/qmk_toolbox/releases) (recommended GUI)

> * [avrdude](http://www.nongnu.org/avrdude/) with usbasp

Reviewer: ```suggestion
* [avrdude](http://www.nongnu.org/avrdude/) with the `usbasp` programmer
```
